sdk/core/Azure.Core/src/Shared/AzureSasCredentialPolicy.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class AzureSasCredentialPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly AzureSasCredential _credential;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureSasCredentialPolicy"/> class.
+        /// </summary>
+        /// <param name="credential">The <see cref="AzureSasCredentialPolicy"/> used to authenticate requests.</param>
+        public AzureSasCredentialPolicy(AzureSasCredential credential)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            _credential = credential;
+        }
+
+        /// <inheritdoc/>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
+            string query = message.Request.Uri.Query;
+            string signature = _credential.Signature;
+            if (signature.StartsWith("?", StringComparison.InvariantCulture))
+            {
+                signature = signature.Substring(1);
```

Comment:
It would be better to use StringBuilder and Spans to minimize allocations in the pipeline if possible.

Reply:
Ouch. In Java `substring` would share internal array with parent. Good to know.

Reply:
I changed the code. However, `StringBuilder.Append` that takes `ReadOnlySpan<string>` isn't available in both netstandard20 and net461 that we target :/ So it isn't as good as it could be in net5... unless there's better way.

Reply:
Looks like we won't get any better with .NET version we're targeting (see comments below). I brought back original code.

Reply:
Have you considered using https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-5.0#System_Text_StringBuilder_Append_System_String_System_Int32_System_Int32_